File: springboot/sara/src/main/java/com/example/sara/service/SensorAssignmentService.java
package com.example.sara.service;

import com.example.sara.dto.SensorAssignmentDto;
import com.example.sara.dto.SensorAssignmentCreateDto;
import java.util.List;

/**
 * Service interface for SensorAssignment operations.
 */
public interface SensorAssignmentService {
    SensorAssignmentDto createSensorAssignment(SensorAssignmentCreateDto dto);
    SensorAssignmentDto updateSensorAssignment(Long id, SensorAssignmentCreateDto dto);
    void deleteSensorAssignment(Long id);
    List<SensorAssignmentDto> getAllSensorAssignments();
}

File: angular/SmartResearchAssistant/src/app/sara-admin/sara-admin/services/sensor-assignment.service.ts
// src/app/sara-admin/services/sensor-assignment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { SensorAssignment } from '../models/sensor-assignment.model';

@Injectable({
  providedIn: 'root'
})
export class SensorAssignmentService {
  private apiUrl = 'http://localhost:8081/api/sensorassignments';

  constructor(private http: HttpClient) { }

  getSensorAssignments(): Observable<SensorAssignment[]> {
    return this.http.get<SensorAssignment[]>(this.apiUrl);
  }

  createSensorAssignment(assignment: { 
    sensorBoxId: number, 
    sensorTypeId: number, 
    displayColor: string 
  }): Observable<SensorAssignment> {
    return this.http.post<SensorAssignment>(this.apiUrl, assignment);
  }

  updateSensorAssignment(id: number, assignment: { 
    sensorBoxId: number, 
    sensorTypeId: number, 
    displayColor: string 
  }): Observable<SensorAssignment> {
    return this.http.put<SensorAssignment>(`${this.apiUrl}/${id}`, assignment);
  }

  deleteSensorAssignment(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}